### PR TITLE
Refinement/work on diff more

### PR DESCRIPTION
**Pull request by tauraamui**
The purpose of these changes are to increase the functional scope of our ability to generate diffs between A + B sets of file lines, and to add tests.

This code is not being used by anything so far, but the intention is this will be used to power our generation of the undo tree's individual nodes.

### CODE REVIEW

File: src/diff.v [hunk 1]
```diff
@@ -55,58 +56,110 @@ fn expand_unique(mut left []Entry, mut right []Entry, dir int) {
...
-	mut r_token := Entry{}
 	mut l_target := 0
 	mut r_target := 0
 
 	for l_pos < lx {
 		l_target = l_pos
 
 		for left[l_target].ref < 0 {
-			l_target += 1
+			l_target++
```
Comment (valxntine) [inline src/diff.v:147]: Why the change from the preferred `+= 1` to `++`?

File: src/diff.v [hunk 2]
```diff
@@ -55,58 +56,110 @@ fn expand_unique(mut left []Entry, mut right []Entry, dir int) {
...
+		}
 	}
 }
 
 fn append_multiple(mut acc []Op, token Entry, kind string) {
 	mut n := token.count
-	for _ in 0..n {
+	for n > 0 {
 		acc << Op{ kind: kind, value: token.value }
+		n--
```
Comment (valxntine) [inline src/diff.v:73]: [nit] if we prefer `+=` over `++` should we prefer `-=` over `--`?

File: src/diff.v (hunk 1 above)
Comment (tauraamui) [reply src/diff.v:147]: No good reason, will change.

File: src/diff.v (hunk 2 above)
Comment (tauraamui) [reply src/diff.v:73]: Probably yes, will change.

**Review by valxntine — APPROVED** [review]
👍🏻
